### ISSUE TITLE
[accessibility] Reset button is missing an accessible name and `role="button"`

### ISSUE THREAD

**Issue by Malvoz** [smeijer/leaflet-geosearch#319]
- The reset button is missing an [_accessible name_](https://www.w3.org/TR/accname-1.1/#dfn-accessible-name), it is currently announced as "times" or "multiplication" depending on the screen reader. It should have [`aria-label="Reset"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label).
- The reset button is a link, though it should be conveyed as a button to screen reader users, using [`role="button"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role).

**Comment by smeijer** [comment]
I'd love to accept a pull request for this.

**Comment by smeijer** [comment]
I'm not assigning tickets. Please just submit the pull if you're feeling for it. Thanks!
